SproxydKeysScan/Logging.js

```diff
@@ -0,0 +1,100 @@
+const werelogs = require('werelogs');
+const { env } = require('./env');
+
+const loggerConfig = {
+    logLevel: env.LOG_LEVEL,
+    dumpLevel: env.DUMP_LEVEL,
+};
+
+werelogs.configure(loggerConfig);
+
+class AggregateLogger {
+    /**
+     * @constructor
+     * @param {number} logInterval - log summary statistics once every intervalSize seconds
+     */
+    constructor(logInterval) {
+        this.log = new werelogs.Logger('s3utils:SproxydKeysScan:AggregateLogger');
+        if (AggregateLogger._instance) {
+            const message = 'AggregateLogger is a singleton and has been instatiated. This instantiation is ignored';
+            AggregateLogger._instance.log.warn(message);
+            return AggregateLogger._instance;
+        }
+        this.interval = logInterval;
+        this.beginTime = new Date();
+        this.statistics = new Map();
+        AggregateLogger._instance = this;
+        return AggregateLogger._instance;
+    }
+
+    update(event) {
+        this.statistics.set(event, (this.statistics.get(event) || 0) + 1);
+    }
+
+    logInterval() {
+        const currentTime = new Date();
+        const summary = {};
+        for (const [k, v] of new Map(this.statistics).entries()) {
```

Comment:
```suggestion
        for (const [k, v] of this.statistics) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#iterating_map_with_for..of